### DISCUSSION TITLE
"Parent" Attribute of magnets

### DISCUSSION THREAD

**Discussion by elafmusa** [atcollab/at#746] [Python AT]
Hi everyone,

I'm currently working with an AT lattice for PETRAIII, that has several quadrupoles connected in series. While examining the quadrupole elements in PyAT, I noticed a common attribute called "Parent."

For example, when looking at two quadrupoles with FamilyName QF and Q2B, both have the same Parent name, which is QA.

Quadrupole:
	FamName : QF
	Length : 0.7028
	PassMethod : StrMPoleSymplectic4Pass
	EApertures : [1. 1.]
	Parent : QA
	NumIntSteps : 9
	MaxOrder : 1
	PolynomA : [0. 0.]
	PolynomB : [0.         0.24013038]
	K : 0.24013038126182484
	
	
Or

Quadrupole:
	FamName : Q2B
	Length : 0.7028
	PassMethod : StrMPoleSymplectic4Pass
	EApertures : [1. 1.]
	Parent : QA
	NumIntSteps : 9
	MaxOrder : 1
	PolynomA : [0. 0.]
	PolynomB : [0.         0.26084005]
	K : 0.26084004702567504

	
I'm curious about the significance of the "Parent" attribute. Does it have any relation to considering multiple quadrupoles within one family? Unfortunately, I couldn't find any information about it in the documentation.

Thank you in advance,
Regards

**Comment by lfarv** [comment]
Hello @elafmusa,

Sorry for the late answer, I was away for a few days.
The "Parent" attribute that you mention is not a standard PyAT attribute. It has probably been added be the user who built the lattice, or by a conversion process from a different format. So this attribute is ignored by PyAT, as any attribute that you are free to add to any element.

You can use such user-defined attributes to identify relationships between elements and process them as you like. More generally, you can use them to record any additional information about elements. But it's up to you to process them.
